app/models/space.rb

```diff
@@ -100,33 +100,42 @@ def potential_duplicates
   # Move this somewhere better, either a service or figure out a way to make it a scope
   # NOTE: this expects a scope for spaces but returns an array
   # preferably we would find some way to return a scope too
-  def self.filter_on_facilities(spaces, facilities) # rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+  def self.filter_on_facilities(spaces, filtered_facilities)
     results = spaces.includes(:space_facilities).filter_map do |space|
-      score = 0
-      space.space_facilities.each do |review|
-        next unless facilities.include?(review.facility_id)
-
-        # The more correct matches the lower the number.
-        # this is so the sort_by later will be correct as it sorts by lowest first
-        # we could do a reverse on the result of sort_by but this will incur
-        # a performance overhead
-        if review.likely?
-          score -= 2
-        elsif review.maybe?
-          score -= 1
-        elsif review.unlikely?
-          score += 1
-        elsif review.impossible?
-          score += 2
-        end
-      end
+      relevant_facilities = space.relevant_facilities
+
+      # If no relevant matches at all, exclude the space:
+      next unless (filtered_facilities & relevant_facilities.map { |sf| sf.facility.id }).any?
 
-      OpenStruct.new(score: score, space: space) # rubocop:disable Style/OpenStructUse
+      space.score_by_filter_on_facilities(filtered_facilities, relevant_facilities)
     end
 
     results.sort_by(&:score).map(&:space)
   end
 
+  def score_by_filter_on_facilities(filtered_facilities, relevant_facilities = self.relevant_facilities)
```

Comment:
Does this need a default value? it is only used by the method above.

Reply:
Probably doesn't! I was just a bit quesy about having it on the Space model and not having it be callable without calling relevant_facilities first. Will probably not be used for anything else, so I guess it could be removed!

Reply:
Removed now.